java/arcs/android/crdt/CrdtParcelables.kt

```diff
@@ -43,11 +43,10 @@ fun Parcel.writeModelData(model: CrdtData?, flags: Int) {
         null -> writeTypedObject(null, flags)
         is CrdtCount.Data ->
             writeTypedObject((model as? CrdtCount.Data)?.toParcelable(), flags)
-        // Caution: CrdtSingleton.Data extends CrdtSet.Data, so we must check it first.
-        is CrdtSingleton.Data<*> ->
-            writeTypedObject((model as? CrdtSingleton.Data<Referencable>)?.toParcelable(), flags)
         is CrdtSet.Data<*> ->
             writeTypedObject((model as? CrdtSet.Data<Referencable>)?.toParcelable(), flags)
+        is CrdtSingleton.Data<*> ->
+            writeTypedObject((model as? CrdtSingleton.Data<Referencable>)?.toParcelable(), flags)
```

Comment:
can you add a simple test to CrdtSingletonTest:
assertThat(alice.data is CrdtSet.Data<*>).isFalse()
with maybe a reference to the emulator bug in a comment

Reply:
Great idea, done!